### DISCUSSION TITLE
How do I exit Bun shell code with a non-zero error code?

### DISCUSSION THREAD

**Discussion by pierskarsenbarg** [oven-sh/bun#9799] [Q&A] [answered]
I don't know if this should be an issue, but what's the best way to do the equivalent of `exit 1`? Context is that I want to run this in a GitHub Action and want it to fail if something bad happens (like a non-zero exit code). _Technically_ it's working because Bun is returning an error code, but it would be useful to be able to use different error codes.

So in bash it would be something like:

```bash
&2 echo "error: bad thing has happened"
exit 1
```

With Bun, I think it should be something like:

```typescript
import { $ } from "bun";

await $`echo "trying to exit with non-zero error code"`
await $`exit 1`
```

But the output from the above is:

```
trying to exit with non-zero error code
bun: command not found: exit
```

**Comment by truongan07** [comment]
```
import { $ } from 'bun';

const fn = num => (0 !== num) ? 0 : 1;

try {
    await $`echo "trying to exit with non-zero error code"`;
    await $`exit ${fn(1)}`;
}catch {
    // do something if parameter of fn is zero
}
```

**Comment by Jarred-Sumner** [comment, marked as answer]
@nektro added support for the `exit` command in bun v1.1

@pierskarsenbarg please try again with bun v1.1
<img width="487" alt="image" src="https://github.com/oven-sh/bun/assets/709451/dcf14c8d-14e6-4813-bd24-5eed3c0c0e18">

**Reply by pierskarsenbarg** [reply]
😂 I was trying to get this working on Sunday juuuust before the 1.1 release. 

Thanks you!
